argonaut/src/main/scala/org/http4s/argonaut/ArgonautInstances.scala

```diff
@@ -16,13 +16,19 @@ trait ArgonautInstances {
       )
     }
 
-  implicit val jsonEncoder: EntityEncoder[Json] =
+  def jsonEncoder(prettyParams: PrettyParams): EntityEncoder[Json] =
     EntityEncoder.stringEncoder(Charset.`UTF-8`).contramap[Json] { json =>
       // TODO naive implementation materializes to a String.
       // Look into replacing after https://github.com/non/jawn/issues/6#issuecomment-65018736
-      Argonaut.nospace.pretty(json)
+      prettyParams.pretty(json)
     }.withContentType(`Content-Type`(MediaType.`application/json`, Charset.`UTF-8`))
 
+  implicit val jsonEncoder: EntityEncoder[Json] =
+    jsonEncoder(Argonaut.nospace)
```

Comment:
Your other idea might look like topic/parameterize-pretty-params-2.  It avoids some overloads, but is a bit weird to define.

The branch, as it stands, is well suited to an ad hoc use of different PrettyParams.  To enforce it globally, one has to be at a higher priority over the jsonEncoderOf implicit.